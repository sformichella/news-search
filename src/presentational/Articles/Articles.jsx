import React from "react";
import PropTypes from "prop-types";
import Article from "../Article copy/Article";

export default function Articles({ articles }) {
  return (
    <ul>
      {
        articles.map((article, i) => {
          return (
            <li key={i}>
              <Article {...article}/>
            </li>
          );
        })
      }
    </ul>
  );
}

Articles.propTypes = {
  articles: PropTypes.array.isRequired
};
