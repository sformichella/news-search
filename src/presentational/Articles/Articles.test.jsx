import React from "react";
import { render, cleanup } from "@testing-library/react";
import Articles from "./Articles";

const articles = [
  { title: "Cool Article 1", author: "Journalist 1", description: "cool 1" },
  { title: "Cool Article 2", author: "Journalist 2", description: "cool 2" },
  { title: "Cool Article 3", author: "Journalist 3", description: "cool 3" }
];

describe("Articles component", () => {
  afterEach(() => cleanup());
  it("renders Articles", () => {
    const { asFragment } = render(<Articles 
      articles={articles}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
