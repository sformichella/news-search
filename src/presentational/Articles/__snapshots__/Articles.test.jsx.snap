// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles component renders Articles 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <h1>
        Cool Article 1
      </h1>
      <h2>
        Journalist 1
      </h2>
      <p>
        cool 1
      </p>
    </li>
    <li>
      <h1>
        Cool Article 2
      </h1>
      <h2>
        Journalist 2
      </h2>
      <p>
        cool 2
      </p>
    </li>
    <li>
      <h1>
        Cool Article 3
      </h1>
      <h2>
        Journalist 3
      </h2>
      <p>
        cool 3
      </p>
    </li>
  </ul>
</DocumentFragment>
`;
